### PR TITLE
Implement webhook support (WIP)

### PR DESCRIPTION
**Pull request by clook**
Implement webhook support for sync orders.
As now, it is configured with a webhook list in the YAML config file similar to task list but with part of task being described only in the event got from webhook as JSON.

Example of config:
```
relay: skopeo

skopeo:
  binary: skopeo
  certs-dir: /etc/skopeo/certs.d

tasks: []

webhooks:
- name: acr2ecr
  verbose: true
  endpoint: /acr2ecr
  source:
    registry: myregistry.azurecr.io
    auth: <base64_docker_credential>
  target:
    registry: 123456789012.dkr.ecr.eu-west-3.amazonaws.com
    auth-refresh: 10h
```

Example of webhook:
```
curl -X POST -d '{"mappings": [{"from": "test/medical/front-e2e", "to": "padoa/medical/front-e2e", "tags": ["e3ff123"]}]}' localhost:8080/acr2ecr
```

- [x] implement HTTP server and webhook
- [ ] <s>refacto / clean code / add tests?</s>
- [x] manage parallel events (go routine)
- [x] improve logging: better tracking of sync (especially if parallel) / report failures
- [ ] <s>avoid multiple parallel syncs for the exact same image (same sha256), it would not benefit layer caching</s>
- [ ] <s>handle gentle termination (ie finish transfer in progress when receiving SIGTERM)</s>
- [ ] <s>document</s>

<s>Striked</s> elements will be handled in a future PR.

### CODE REVIEW

File: internal/pkg/sync/config.go [hunk 1]
```diff
@@ -198,19 +257,34 @@ func (t *task) fail(f bool) {
...
 	}
 	return from, to
 }
 
-//
-func (t *task) ensureTargetExists(ref string) error {
+func (w *webhook) mappingRefs(m *mapping) (err error, from string, to string) {
+	if m != nil {
+		if !strings.HasPrefix(m.From, w.FromPrefix) {
+			return fmt.Errorf("Prefix \"%s\" not found in from mapping \"%s\"", m.From, w.FromPrefix), "", ""
```
Comment (clook) [inline internal/pkg/sync/config.go:276]: Prefix and from mapping are inverted.

File: internal/pkg/relays/skopeo/skopeo.go [hunk 2]
```diff
@@ -81,12 +81,7 @@ func chooseOutStream(out io.Writer, verbose, isErrorStream bool) io.Writer {
 		if out != nil {
 			return out
 		}
-		if log.ToTerminal {
-			if isErrorStream {
-				return os.Stderr
-			}
-			return os.Stdout
-		}
+		return os.Stdout
```
Comment (RaJiska) [inline internal/pkg/relays/skopeo/skopeo.go:84]: Return stdout / stderr depending of the parameter

**Review by RaJiska — COMMENTED** [review]
Skopeo relay structure is now legacy.

File: internal/pkg/sync/config.go (hunk 1 above)
Comment (clook) [reply internal/pkg/sync/config.go]: fixed.

**Comment by clook** [comment]
Scope has been reduced to minimum viable. PR ready and paired review.

**Pull request merged**